### PR TITLE
[Feature:InstructorUI/RainbowGrades] Initial work on automatic rainbow grades

### PR DESCRIPTION
**Pull request by Kevin-MM**
@holzbh started this feature and most credit for the user interface goes to him

### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant [#123](https://github.com/Submitty/submitty.github.io/pull/123)

### What is the current behavior?
Previously there was no user interface for customization / generation of rainbow grades.

### What is the new behavior
When logged in as a course admin you will now see a 'Web-Based Rainbow Grades Generation' under the 'Grade Reports' tab.  This form allows for some gui based configuration of rainbow grades.  When the 'Save Changes' button is clicked rainbow grades will be automatically generated or if this form has already been used in the past then new changes will be applied.

**How to use**
- First ensure the [jobs daemon is running](https://submitty.org/developer/jobs_daemon) by calling ```sudo systemctl start submitty_daemon_jobs_handler``` prior to running the submitty install script
- Ensure the RainbowGrades repo is cloned in your GIT_CHECKOUT directory
- Log into the 'sample'
- Press the 'Generate Grade Summaries' button inside 'Grade Reports' tab
- Press the 'Web-Based Rainbow Grades Generation' button
- Play with the new customization user interface, saving the form when satisfied
- Check out the results in the 'Rainbow Grades' tab (must be enabled inside the course configuration page)

**Limitations**
- Currently grade summaries must be generated manually before this feature can work
- The 'points distribution' part of gradeables isn't currently implemented
- The first time this form is used, it takes a minute to generate rainbow grades.  The is currently no feedback about when this process has completed (fixed in a future PR)
- The system breaks if you submit the form with a blank messages box (fixed in a future PR)

**Other notes**
- An instructor may place a 'custom_customization.json' file inside their rainbow_grades folder.  This file will always take priority over anything generated by the web user interface and functionality of the web user interface is undefined in this case.
- Output of auto_rainbow_grades.py is captured in 'auto_debug_output.txt' of the course's rainbow_grades directory

### Other information?
- This PR should not break anything, appropriate migrators are included.
- Tested manually

### CODE REVIEW

**Comment by Kevin-MM** [comment]
Partially Addresses #3745

**Comment by bmcutler** [comment]
If I leave the "Messages" box empty, I get an "Internal Server Error" popup.  Besides being a cryptic message, this should be optional field.

**Comment by bmcutler** [comment]
If I've saved the "Rainbow Grades Customization" page at least once, it says "Saved successfully".  But then if I continue to edit the form that message does not clear (confusing, since the new edits aren't saved).

File: sbin/auto_rainbow_grades.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+"""
+
+# Imports
+import sys
+import os
+import shutil
+import pwd
+
+# Constants
+RAINBOW_GRADES_PATH = '/usr/local/submitty/GIT_CHECKOUT/RainbowGrades'
```
Comment (bmcutler) [inline sbin/auto_rainbow_grades.py]: paths /usr/local/submitty, /var/local/submitty, and user submitty_php should be read from config files
/usr/local/submitty/config/submitty.json 
/usr/local/submitty/config/submitty_users.json

(and you can get to those config files from ../.. etc from the relative location of the installed script)

File: sbin/auto_rainbow_grades.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+COURSES_PATH = '/var/local/submitty/courses'
+DEFAULT_USER = 'submitty_php'
```
Comment (bmcutler) [inline sbin/auto_rainbow_grades.py]: actually I think we should use the submitty_daemon user here.

**Comment by Kevin-MM** [comment]
Removed hard coded paths, and changed user to submitty_daemon user in most recent push.

**Pull request merged**